data/types/base.go

```diff
@@ -197,11 +199,10 @@ func (b *Base) Validate(validator structure.Validator) {
 		stringValidator.Exists().NotEmpty().LengthLessThanOrEqualTo(TagLengthMaximum)
 	}).EachUnique()
 
-	timeValidator := validator.String("time", b.Time)
+	timeValidator := validator.Time("time", b.Time)
 	if b.Type != "upload" { // HACK: Need to replace upload.Upload with data.DataSet
-		timeValidator.Exists()
+		timeValidator.Exists().NotZero()
```

Comment:
Is there a reason there is a `NotZero()` condition here? The time field is pointer to time, thus it's empty value is `nil`.

Reply:
i started with not-pointers, corrected.

Reply:
I just saw that you had to remove `timeValidator.AsTime(TimeFormat)` which is roughly equivalent to the `NotZero()` check here.